src/mqttcm_connect.c

```diff
@@ -1573,6 +1588,78 @@ int isSubscribeNeeded(char *compname)
 	return 2;
 }
 
+char* GetTopicFromSubcribeId(int subscribeId)
+{
+	comp_topic_name_t* temp = g_head;
+	while (temp != NULL)
+	{
+		if(temp->subscribeId == subscribeId)
+		{
+			return (temp->compName);
+		}
+		temp = temp->next;
+	}
+	return NULL;
+}
+
+void AddSubscriptionIdToList(char *comp, int subscribeId)
+{
+	comp_topic_name_t* temp = g_head;
+	while (temp != NULL)
+	{
+		if(strcmp(temp->compName, comp) == 0)
+		{
+			temp->subscribeId = subscribeId;
+			return;
+		}
+		temp = temp->next;
+	}
+}
+
+void insert(char * compName, char* topic)
+{
+	comp_topic_name_t* newNode = (comp_topic_name_t*)malloc(sizeof(comp_topic_name_t));
+	if(newNode)
+	{
+		memset(newNode, 0, sizeof(comp_topic_name_t) );
+		strncpy(newNode->compName, compName, sizeof(newNode->compName) - 1);
```

Comment:
Null check is needed in insert func also . But do we need insert() func, insert() and AddToSubscriptionList are same right? Can we not reuse the same AddToSubscriptionList() , so that node update will happen at single place ?

Reply:
AddToSubscriptionList add NULL check is removed? it is required.

Reply:
Null checks are added in the new review